x/did/types/did.go

```diff
@@ -0,0 +1,436 @@
+package types
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/btcsuite/btcutil/base58"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	DIDMethod     = "panacea"
+	Base58Charset = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
+)
+
+type SignableDID string
+
+// GetSignBytes returns a byte array which is used to generate a signature for verifying DID ownership.
+func (did SignableDID) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.Amino.MustMarshalJSON(did))
+}
+
+func NewDID(pubKey []byte) string {
+	hash := sha256.New()
+	_, err := hash.Write(pubKey)
+	if err != nil {
+		panic("failed to calculate SHA256 for DID")
+	}
+	idStr := base58.Encode(hash.Sum(nil))
+	return fmt.Sprintf("did:%s:%s", DIDMethod, idStr)
+}
+
+func ParseDID(str string) (string, error) {
+	did := str
+	if !ValidDID(did) {
+		return "", Error(ErrInvalidDID, str)
+	}
+	return did, nil
+}
+
+func ValidDID(did string) bool {
```

Comment:
For consistency, I think the function name `ValidateDID` is better (which is verb-prefixed).
Previously, it was adjective which represent a property of `DID`. But now, I think verb is better :).